asf_search/WKT/validate_wkt.py

```diff
@@ -0,0 +1,330 @@
+import logging
+from numbers import Number
+from typing import Union, Tuple, List
+from shapely import wkt
+from shapely.geometry.base import BaseGeometry
+from shapely.geometry import Polygon, MultiPolygon, Point, MultiPoint, LineString, MultiLineString, GeometryCollection
+from shapely.geometry.collection import BaseMultipartGeometry
+from shapely.geometry.polygon import orient
+from shapely.ops import transform, orient, unary_union
+from .RepairEntry import RepairEntry
+from sklearn.neighbors import NearestNeighbors
+import numpy as np
+
+from asf_search.exceptions import ASFWKTError
+
+
+def validate_wkt(aoi: Union[str, BaseGeometry]) -> BaseGeometry:
+    """
+    Param aoi: the WKT string or Shapely Geometry to validate and prepare for the CMR query
+    Validates the given area of interest, and returns a validated and simplified WKT string
+    returns: The input AOI's CMR ready WKT string
+    """    
+    aoi_shape = BaseGeometry()
+
+    if isinstance(aoi, str):
+        aoi_shape = wkt.loads(aoi)
+    else:
+        aoi_shape = wkt.loads(aoi.wkt)
+
+    if not aoi_shape.is_valid:
+        aoi_shape = _search_wkt_prep(aoi_shape)
```

Comment:
For [.is_valid](https://shapely.readthedocs.io/en/stable/manual.html#object.is_valid), it looks like only polygons and multi-polygons might return False? And then, multi-polygons with overlapping poly's are not considered valid.

This also means a lot of code in `._search_wkt_prep` might not get hit if only polygons are passed though? I'm just confused on what this part does, but I probably miss-read something.